operator/internal/controller/redpanda/redpanda_controller.go

```diff
@@ -211,206 +225,248 @@ func (r *RedpandaReconciler) Reconcile(ctx context.Context, req ctrl.Request) (r
+	if r.UseNodePools {
+		existingPools, err = fromSourceCluster(ctx, r.Client, "pool", rp, &redpandav1alpha2.NodePoolList{})
+		if err != nil {
+			logger.Error(err, "fetching desired node pools")
+			return nil, err
+		}
 	}
 
-	// next we sync up all of our pools themselves
-	requeue, err := r.reconcilePools(ctx, cluster, pools)
-	if err != nil {
-		status.Status.SetResourcesSynced(statuses.ClusterResourcesSyncedReasonError, err.Error())
+	rp.ManagedFields = nil // nil out our managed fields
+	cluster := lifecycle.NewClusterWithPools(rp, existingPools...)
 
-		logger.Error(err, "error reconciling pools")
-		return r.syncStatusErr(ctx, err, status, cluster)
-	}
-	if requeue || !pools.AnyReady() {
-		// we might have no brokers ready at this point, so we can't do anything below this
-		// since they require the ability to talk to our brokers
-		return r.syncStatusAndRequeue(ctx, status, cluster)
+	// grab our existing and desired pool resources
+	// so that we can immediately calculate cluster status
+	// from and sync in any subsequent operation that
+	// early returns
+	restartOnConfigChange := feature.RestartOnConfigChange.Get(ctx, rp)
+	injectedConfigVersion := ""
+	if restartOnConfigChange {
+		injectedConfigVersion = rp.Status.ConfigVersion
 	}
-
-	admin, err := r.ClientFactory.RedpandaAdminClient(ctx, rp)
+	pools, err := r.LifecycleClient.FetchExistingAndDesiredPools(ctx, cluster, injectedConfigVersion)
 	if err != nil {
-		logger.Error(err, "error fetching redpanda admin client")
-		return r.syncStatusErr(ctx, err, status, cluster)
+		logger.Error(err, "fetching pools")
+		return nil, err
 	}
-	defer admin.Close()
-
-	// now we ensure that we reconcile all of our decommissioning nodes
-	// TODO: Do we want to rate limit this as well given that it also calls the admin API?
-	// My thought is no since we want to be snappy with decommissioning.
-	health, requeue, err := r.reconcileDecommission(ctx, cluster, admin, pools)
-	if err != nil {
-		status.Status.SetResourcesSynced(statuses.ClusterResourcesSyncedReasonError, err.Error())
-		if err != nil {
-			if internalclient.IsTerminalClientError(err) {
-				status.Status.SetHealthy(statuses.ClusterHealthyReasonTerminalError, err.Error())
-			} else {
-				status.Status.SetHealthy(statuses.ClusterHealthyReasonError, err.Error())
-			}
-		}
 
-		logger.Error(err, "error decommissioning brokers")
-		return r.syncStatusErr(ctx, err, status, cluster)
-	}
-	if requeue {
-		return r.syncStatusAndRequeue(ctx, status, cluster)
-	}
+	status := lifecycle.NewClusterStatus()
+	status.Pools = pools.PoolStatuses()
 
-	status.Status.SetResourcesSynced(statuses.ClusterResourcesSyncedReasonSynced)
-	if health.IsHealthy {
-		status.Status.SetHealthy(statuses.ClusterHealthyReasonHealthy)
+	if pools.AnyReady() {
+		status.Status.SetReady(statuses.ClusterReadyReasonReady)
+	} else if pools.AllZero() {
+		status.Status.SetReady(statuses.ClusterReadyReasonReady, messageNoBrokers)
```

Comment:
oooooh, I understand the test below now. What's the rationale for this OOC? I would expect the cluster to report as unready in this case because there's nothing to be ready.